### PR TITLE
HiveConnection pathPrefix optional, FileIncrementalMoveMode absolute archivePath

### PR DESCRIPTION
**Pull request by zzeekk**
### What changes are included in the pull request?
see #540, #558

Note that FileIncrementalMoveMode.archiveSubdirectory is renamed as archivePath.
This is not a breaking change as it is only present in the current 2.3.0-SNAPSHOT version.

### CODE REVIEW

File: sdl-core/src/test/scala/io/smartdatalake/workflow/action/ExecutionModeTest.scala [hunk 1]
```diff
@@ -208,9 +209,40 @@ class ExecutionModeTest extends FunSuite with BeforeAndAfter {
     intercept[NoDataToProcessWarning](executionMode.apply(ActionId("test"), fileEmptyDO, fileEmptyDO, subFeed, PartitionValues.oneToOneMapping))
   }
 
-  test("FileIncrementalMoveMode create archive filename") {
-    val executionMode = FileIncrementalMoveMode(archiveSubdirectory = Some("_archive"))
-    assert(executionMode.createArchiveFileName("hdfs://nameservice1/test/abc.csv") == "hdfs://nameservice1/test/_archive/abc.csv")
+  test("FileIncrementalMoveMode archive relative path") {
+    val srcDOArchive = ParquetFileDataObject( "srcArchive", tempPath+s"/srcArchive")
```
Comment (Geheiner) [inline sdl-core/src/test/scala/io/smartdatalake/workflow/action/ExecutionModeTest.scala]: do you need the +s here?

File: sdl-core/src/test/scala/io/smartdatalake/workflow/action/ExecutionModeTest.scala [hunk 2]
```diff
@@ -208,9 +209,40 @@ class ExecutionModeTest extends FunSuite with BeforeAndAfter {
...
+
+    val executionMode = FileIncrementalMoveMode(archivePath = Some("archive"))
+    executionMode.prepare(ActionId("test"))
+    val subFeed: FileSubFeed = FileSubFeed(fileRefs = None, dataObjectId = srcDOArchive.id, partitionValues = Seq())
+    val result = executionMode.apply(ActionId("test"), srcDOArchive, srcDOArchive, subFeed, PartitionValues.oneToOneMapping).get
+    val subFeedWithFileRefs = subFeed.copy(fileRefs = result.fileRefs, fileRefMapping = result.fileRefs.map(_.map(fileRef => FileRefMapping(fileRef,fileRef))))
+    assert(srcDOArchive.getFileRefs(Seq()).nonEmpty)
+    executionMode.postExec(ActionId("test"), srcDOArchive, srcDOArchive, subFeedWithFileRefs, subFeedWithFileRefs)
+    assert(srcDOArchive.getFileRefs(Seq()).isEmpty)
+    assert(srcDOArchive.filesystem.listStatus(new Path(srcDOArchive.path+s"/archive")).toSeq.nonEmpty)
```
Comment (Geheiner) [inline sdl-core/src/test/scala/io/smartdatalake/workflow/action/ExecutionModeTest.scala]: do you need the +s here?

File: sdl-core/src/test/scala/io/smartdatalake/workflow/action/ExecutionModeTest.scala [hunk 3]
```diff
@@ -208,9 +209,40 @@ class ExecutionModeTest extends FunSuite with BeforeAndAfter {
...
+  }
+
+  test("FileIncrementalMoveMode archive absolute path") {
+    val srcDOArchive = ParquetFileDataObject( "srcArchive", tempPath+"/srcArchive")
```
Comment (Geheiner) [inline sdl-core/src/test/scala/io/smartdatalake/workflow/action/ExecutionModeTest.scala]: run the formatter in IntelliJ for this file. Afterwards, it should look tempPath + "/srcArchive"

**Review by Geheiner — APPROVED** [review]

**Comment by Geheiner** [comment]
LGTM

File: sdl-core/src/test/scala/io/smartdatalake/workflow/action/ExecutionModeTest.scala (hunk 1 above)
Comment (zzeekk) [reply sdl-core/src/test/scala/io/smartdatalake/workflow/action/ExecutionModeTest.scala]: fixed

File: sdl-core/src/test/scala/io/smartdatalake/workflow/action/ExecutionModeTest.scala (hunk 3 above)
Comment (zzeekk) [reply sdl-core/src/test/scala/io/smartdatalake/workflow/action/ExecutionModeTest.scala]: fixed

File: sdl-core/src/test/scala/io/smartdatalake/workflow/action/ExecutionModeTest.scala (hunk 2 above)
Comment (zzeekk) [reply sdl-core/src/test/scala/io/smartdatalake/workflow/action/ExecutionModeTest.scala]: fixed

**Pull request merged**